### PR TITLE
Improve web and node configs

### LINKED ISSUE #3: Inline environment variables opt in
Originally raised by @macdonaldr93 in the old repo ([here](https://github.com/Shopify/javascript/issues/252))

> Working a project where I don't want process.env to be automatically transformed into value. Due to babel-plugin-transform-inline-environment-variables, my environment variables are getting transformed to undefined.
>
> https://github.com/Shopify/javascript/blob/master/packages/babel-preset-shopify/non-standard-features.js#L7

### LINKED ISSUE #4: Use babel-preset-env
Originally raised by @lemonmade in the JS repo: [here](https://github.com/Shopify/javascript/issues/250)

> Seems like a clear win in place of the es2015, es2016, and es2017 presets.
>
> https://github.com/babel/babel-preset-env

### PR DESCRIPTION
**Pull request by lemonmade**
Fixes #3
Fixes #4 

This PR improves our Babel configs substantially. It also introduces a few breaking changes:

- Both the `web` and `node` configs rely on `babel-preset-env`, so they automatically transpile features based on your target environment. For `node`, this defaults to the current node version (configurable with the `version` option). For `web`, this defaults to a browserlist stolen from core (configurable with the `browsers` option).
- Use built ins (usually `Object.assign`) wherever possible
- Removed stage 2 as a default set that's included (default to stage 3), but kept the class properties transform. I think this is a more sensible policy.
- Made inlining env variables opt-in (instead of always being on).
- Added the runtime transform, which delegates helper functions/ regenerator to `babel-runtime` instead of inlining for every file.

cc/ @Shopify/javascript

### CODE REVIEW

File: web.js [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
   options = options || {};
+  var modules = options.modules == null ? true : options.modules;
 
   return {
     presets: [
-      [require('babel-preset-es2015').buildPreset, options],
-      require('babel-preset-es2016'),
-      require('babel-preset-es2017'),
-      require('./non-standard-features'),
+      [require.resolve('babel-preset-env'), {
```
Comment (GoodForOneFare) [inline web.js:10]: Squeeeeeeeeeeeeeee! 😸

File: README.md [hunk 2]
```diff
@@ -36,32 +36,57 @@ This packages comes with several different presets for you to use, depending on
...
+          "presets": {
+            ["shopify/web", {
+              "browsers": ["last 3 versions"]
+            }]
+          }
+        }
+      }
+      ```
+    
+    - `inlineEnv`, a boolean (defaults to `false`) to automatically replace `process.env.<VAR>` statements with the corresponding environment variable.
```
Comment (macdonaldr93) [inline README.md:69]: 🎉   stoked on this being opt-in.

File: flow.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   plugins: [
-    require('babel-plugin-syntax-flow'),
-    require('babel-plugin-transform-flow-strip-types'),
+    require.resolve('babel-plugin-syntax-flow'),
```
Comment (GoodForOneFare) [inline flow.js:3]: `require.resolve` gets the absolute path so that libs are loaded only when they're needed?

File: node.js [hunk 4]
```diff
@@ -1,43 +1,21 @@
-var semver = require('semver');
+var nonStandardPlugins = require('./non-standard-plugins');
 
 module.exports = function shopifyNodePreset(context, options) {
   options = options || {};
-  var version = options.version || process.version;
+  var version = options.version || true;
```
Comment (GoodForOneFare) [inline node.js]: Can we use `"current"` here, please?  Seems to [do the same thing](https://github.com/babel/babel-preset-env#targetsnode), and looks more explicit to me.

**Review by ismail-syed — APPROVED** [review]
👍

**Review by GoodForOneFare — APPROVED** [review]
A big step forward.

File: flow.js (hunk 3 above)
Comment (lemonmade) [reply flow.js:3]: Yup :)

**Pull request merged**